## t5-base + REALEC

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt


=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
351	894	605	0.2819	0.3672	0.2957



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {1}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M              74       101      162      0.4229   0.3136   0.3953
R              252      658      393      0.2769   0.3907   0.294
U              25       135      50       0.1562   0.3333   0.1748

=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
351	894	605	0.2819	0.3672	0.2957



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {2}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            9        11       4        0.45     0.6923   0.4839
ADJ:FORM       0        3        1        0.0      0.0      0.0
ADV            5        7        3        0.4167   0.625    0.4464
CONJ           2        0        0        1.0      1.0      1.0
DET            58       123      177      0.3204   0.2468   0.3024
MORPH          13       24       14       0.3514   0.4815   0.3714
NOUN           38       128      53       0.2289   0.4176   0.2517
NOUN:INFL      0        0        1        1.0      0.0      0.0
NOUN:NUM       11       46       31       0.193    0.2619   0.2037
NOUN:POSS      0        1        0        0.0      1.0      0.0
ORTH           4        132      33       0.0294   0.1081   0.0344
OTHER          57       148      58       0.278    0.4957   0.3048
PART           1        5        1        0.1667   0.5      0.1923
PREP    

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {3}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M:ADJ          0        3        0        0.0      1.0      0.0
M:ADV          0        3        0        0.0      1.0      0.0
M:DET          43       40       135      0.5181   0.2416   0.4216
M:NOUN         6        8        5        0.4286   0.5455   0.4478
M:OTHER        10       12       4        0.4545   0.7143   0.4902
M:PART         0        1        0        0.0      1.0      0.0
M:PREP         4        20       6        0.1667   0.4      0.1887
M:PRON         0        1        0        0.0      1.0      0.0
M:PUNCT        6        3        5        0.6667   0.5455   0.6383
M:VERB         4        5        5        0.4444   0.4444   0.4444
M:VERB:TENSE   1        5        2        0.1667   0.3333   0.1852
R:ADJ          8        8        4        0.5      0.6667   0.5263
R:ADJ:FORM     0        3        1        0.0      0.0      0.0
R:ADV      

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds


============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
392	853	564	0.3149	0.41	0.3302



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {1}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M              80       98       156      0.4494   0.339    0.4219
R              283      632      362      0.3093   0.4388   0.3287
U              29       123      46       0.1908   0.3867   0.2123

============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
392	853	564	0.3149	0.41	0.3302



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {2}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            9        10       4        0.4737   0.6923   0.5056
ADJ:FORM       0        3        1        0.0      0.0      0.0
ADV            6        7        2        0.4615   0.75     0.5
CONJ           2        0        0        1.0      1.0      1.0
DET            63       113      172      0.358    0.2681   0.3355
MORPH          18       24       9        0.4286   0.6667   0.4615
NOUN           42       124      49       0.253    0.4615   0.2781
NOUN:INFL      0        0        1        1.0      0.0      0.0
NOUN:NUM       14       45       28       0.2373   0.3333   0.2518
NOUN:POSS      0        1        0        0.0      1.0      0.0
ORTH           6        132      31       0.0435   0.1622   0.0509
OTHER          65       138      50       0.3202   0.5652   0.3506
PART           1        5        1        0.1667   0.5      0.1923
PREP       

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {3}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M:ADJ          0        3        0        0.0      1.0      0.0
M:ADV          0        3        0        0.0      1.0      0.0
M:DET          47       40       131      0.5402   0.264    0.4468
M:NOUN         7        8        4        0.4667   0.6364   0.493
M:OTHER        11       11       3        0.5      0.7857   0.5392
M:PART         0        1        0        0.0      1.0      0.0
M:PREP         4        19       6        0.1739   0.4      0.1961
M:PRON         0        1        0        0.0      1.0      0.0
M:PUNCT        6        2        5        0.75     0.5455   0.6977
M:VERB         4        5        5        0.4444   0.4444   0.4444
M:VERB:TENSE   1        5        2        0.1667   0.3333   0.1852
R:ADJ          8        7        4        0.5333   0.6667   0.5556
R:ADJ:FORM     0        3        1        0.0      0.0      0.0
R:ADV       

## t5-base + Llama

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt


=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
445	890	2266	0.3333	0.1641	0.2764



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {1}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M              57       149      352      0.2767   0.1394   0.2311
R              325      595      1611     0.3533   0.1679   0.2894
U              63       146      303      0.3014   0.1721   0.2621

=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
445	890	2266	0.3333	0.1641	0.2764



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {2}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            20       33       42       0.3774   0.3226   0.365
ADJ:FORM       1        1        1        0.5      0.5      0.5
ADV            9        21       33       0.3      0.2143   0.2778
CONJ           6        5        7        0.5455   0.4615   0.5263
DET            31       67       315      0.3163   0.0896   0.21
MORPH          7        14       45       0.3333   0.1346   0.2574
NOUN           76       146      305      0.3423   0.1995   0.2994
NOUN:NUM       6        20       66       0.2308   0.0833   0.1705
ORTH           3        3        47       0.5      0.06     0.2027
OTHER          170      363      703      0.3189   0.1947   0.2829
PART           2        5        9        0.2857   0.1818   0.2564
PREP           41       81       237      0.3361   0.1475   0.2676
PRON           2        10       17       0.1667   0.1053   0.1493
PU

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {3}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M:ADJ          3        7        7        0.3      0.3      0.3
M:ADV          1        4        8        0.2      0.1111   0.1724
M:CONJ         1        2        2        0.3333   0.3333   0.3333
M:DET          9        23       190      0.2812   0.0452   0.1376
M:NOUN         9        19       24       0.3214   0.2727   0.3103
M:OTHER        18       54       48       0.25     0.2727   0.2542
M:PART         0        1        0        0.0      1.0      0.0
M:PREP         6        14       39       0.3      0.1333   0.24
M:PRON         0        3        2        0.0      0.0      0.0
M:PUNCT        2        2        1        0.5      0.6667   0.5263
M:VERB         5        12       20       0.2941   0.2      0.2688
M:VERB:FORM    0        1        0        0.0      1.0      0.0
M:VERB:TENSE   3        7        11       0.3      0.2143   0.2778
R:ADJ     

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds


============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
624	711	2087	0.4674	0.2302	0.3875



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {1}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M              63       142      346      0.3073   0.154    0.2563
R              478      452      1458     0.514    0.2469   0.4226
U              83       117      283      0.415    0.2268   0.3559

============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
624	711	2087	0.4674	0.2302	0.3875



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {2}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            25       26       37       0.4902   0.4032   0.4699
ADJ:FORM       1        1        1        0.5      0.5      0.5
ADV            14       17       28       0.4516   0.3333   0.4217
CONJ           8        3        5        0.7273   0.6154   0.7018
DET            42       53       304      0.4421   0.1214   0.2893
MORPH          13       10       39       0.5652   0.25     0.4514
NOUN           101      123      280      0.4509   0.2651   0.3955
NOUN:NUM       9        12       63       0.4286   0.125    0.2885
ORTH           4        3        46       0.5714   0.08     0.2564
OTHER          242      293      631      0.4523   0.2772   0.4016
PART           2        4        9        0.3333   0.1818   0.2857
PREP           63       58       215      0.5207   0.2266   0.4134
PRON           3        8        16       0.2727   0.1579   0.2381

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {3}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M:ADJ          3        7        7        0.3      0.3      0.3
M:ADV          1        4        8        0.2      0.1111   0.1724
M:CONJ         1        2        2        0.3333   0.3333   0.3333
M:DET          12       23       187      0.3429   0.0603   0.177
M:NOUN         9        19       24       0.3214   0.2727   0.3103
M:OTHER        20       48       46       0.2941   0.303    0.2959
M:PART         0        1        0        0.0      1.0      0.0
M:PREP         7        13       38       0.35     0.1556   0.28
M:PRON         0        3        2        0.0      0.0      0.0
M:PUNCT        2        2        1        0.5      0.6667   0.5263
M:VERB         5        12       20       0.2941   0.2      0.2688
M:VERB:FORM    0        1        0        0.0      1.0      0.0
M:VERB:TENSE   3        7        11       0.3      0.2143   0.2778
R:ADJ      

## t5-base + GED + GEC + REALEC

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt


=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
414	543	542	0.4326	0.4331	0.4327



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {1}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M              76       73       160      0.5101   0.322    0.4567
R              310      387      335      0.4448   0.4806   0.4515
U              28       83       47       0.2523   0.3733   0.2697

=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
414	543	542	0.4326	0.4331	0.4327



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {2}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            8        14       5        0.3636   0.6154   0.396
ADJ:FORM       0        1        1        0.0      0.0      0.0
ADV            6        7        2        0.4615   0.75     0.5
CONJ           2        1        0        0.6667   1.0      0.7143
DET            61       69       174      0.4692   0.2596   0.404
MORPH          12       18       15       0.4      0.4444   0.4082
NOUN           46       98       45       0.3194   0.5055   0.3448
NOUN:INFL      0        0        1        1.0      0.0      0.0
NOUN:NUM       14       22       28       0.3889   0.3333   0.3763
ORTH           12       11       25       0.5217   0.3243   0.4651
OTHER          72       193      43       0.2717   0.6261   0.3064
PART           1        1        1        0.5      0.5      0.5
PREP           35       36       18       0.493    0.6604   0.5193
PRON      

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {3}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M:ADJ          0        1        0        0.0      1.0      0.0
M:ADV          0        1        0        0.0      1.0      0.0
M:DET          41       22       137      0.6508   0.2303   0.4767
M:NOUN         7        11       4        0.3889   0.6364   0.4217
M:OTHER        8        21       6        0.2759   0.5714   0.3077
M:PREP         5        6        5        0.4545   0.5      0.463
M:PUNCT        7        4        4        0.6364   0.6364   0.6364
M:VERB         6        6        3        0.5      0.6667   0.5263
M:VERB:TENSE   2        1        1        0.6667   0.6667   0.6667
R:ADJ          7        12       5        0.3684   0.5833   0.3977
R:ADJ:FORM     0        1        1        0.0      0.0      0.0
R:ADV          2        4        0        0.3333   1.0      0.3846
R:CONJ         1        0        0        1.0      1.0      1.0
R:DET    

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds


============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
443	514	513	0.4629	0.4634	0.463



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {1}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M              79       71       157      0.5267   0.3347   0.4725
R              332      365      313      0.4763   0.5147   0.4835
U              32       78       43       0.2909   0.4267   0.3107

============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
443	514	513	0.4629	0.4634	0.463



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {2}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            9        13       4        0.4091   0.6923   0.4455
ADJ:FORM       0        1        1        0.0      0.0      0.0
ADV            6        7        2        0.4615   0.75     0.5
CONJ           2        1        0        0.6667   1.0      0.7143
DET            64       64       171      0.5      0.2723   0.4284
MORPH          15       18       12       0.4545   0.5556   0.4717
NOUN           49       96       42       0.3379   0.5385   0.3651
NOUN:INFL      0        0        1        1.0      0.0      0.0
NOUN:NUM       16       21       26       0.4324   0.381    0.4211
ORTH           13       10       24       0.5652   0.3514   0.5039
OTHER          77       182      38       0.2973   0.6696   0.3345
PART           2        1        0        0.6667   1.0      0.7143
PREP           35       35       18       0.5      0.6604   0.5255
PRON 

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {3}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M:ADJ          0        1        0        0.0      1.0      0.0
M:ADV          0        1        0        0.0      1.0      0.0
M:DET          43       22       135      0.6615   0.2416   0.4909
M:NOUN         8        11       3        0.4211   0.7273   0.4598
M:OTHER        8        20       6        0.2857   0.5714   0.3175
M:PREP         5        6        5        0.4545   0.5      0.463
M:PUNCT        7        3        4        0.7      0.6364   0.6863
M:VERB         6        6        3        0.5      0.6667   0.5263
M:VERB:TENSE   2        1        1        0.6667   0.6667   0.6667
R:ADJ          8        11       4        0.4211   0.6667   0.4545
R:ADJ:FORM     0        1        1        0.0      0.0      0.0
R:ADV          2        4        0        0.3333   1.0      0.3846
R:CONJ         1        0        0        1.0      1.0      1.0
R:DET    

## t5-base + GED + GEC + Llama

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {1}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M              103      30       133      0.7744   0.4364   0.6706
R              365      130      280      0.7374   0.5659   0.6952
U              38       42       37       0.475    0.5067   0.481

=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
506	202	450	0.7147	0.5293	0.6679



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {2}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            12       3        1        0.8      0.9231   0.8219
ADJ:FORM       0        3        1        0.0      0.0      0.0
ADV            8        0        0        1.0      1.0      1.0
CONJ           2        0        0        1.0      1.0      1.0
DET            75       59       160      0.5597   0.3191   0.4864
MORPH          20       10       7        0.6667   0.7407   0.6803
NOUN           59       29       32       0.6705   0.6484   0.6659
NOUN:INFL      0        0        1        1.0      0.0      0.0
NOUN:NUM       19       17       23       0.5278   0.4524   0.5108
ORTH           12       5        25       0.7059   0.3243   0.5714
OTHER          93       25       22       0.7881   0.8087   0.7922
PART           1        1        1        0.5      0.5      0.5
PREP           43       10       10       0.8113   0.8113   0.8113
PRON       

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {3}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M:DET          54       19       124      0.7397   0.3034   0.5745
M:NOUN         8        3        3        0.7273   0.7273   0.7273
M:OTHER        13       2        1        0.8667   0.9286   0.8784
M:PREP         7        3        3        0.7      0.7      0.7
M:PUNCT        10       1        1        0.9091   0.9091   0.9091
M:VERB         9        1        0        0.9      1.0      0.9184
M:VERB:TENSE   2        1        1        0.6667   0.6667   0.6667
R:ADJ          11       3        1        0.7857   0.9167   0.8088
R:ADJ:FORM     0        3        1        0.0      0.0      0.0
R:ADV          2        0        0        1.0      1.0      1.0
R:CONJ         1        0        0        1.0      1.0      1.0
R:DET          10       5        6        0.6667   0.625    0.6579
R:MORPH        20       10       7        0.6667   0.7407   0.6803
R:NOUN  

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds


============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
519	189	437	0.7331	0.5429	0.6851



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {1}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M              105      29       131      0.7836   0.4449   0.6801
R              375      121      270      0.756    0.5814   0.7132
U              39       39       36       0.5      0.52     0.5039

============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
519	189	437	0.7331	0.5429	0.6851



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {2}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            12       3        1        0.8      0.9231   0.8219
ADJ:FORM       0        3        1        0.0      0.0      0.0
ADV            8        0        0        1.0      1.0      1.0
CONJ           2        0        0        1.0      1.0      1.0
DET            76       57       159      0.5714   0.3234   0.4954
MORPH          20       10       7        0.6667   0.7407   0.6803
NOUN           61       29       30       0.6778   0.6703   0.6763
NOUN:INFL      0        0        1        1.0      0.0      0.0
NOUN:NUM       20       17       22       0.5405   0.4762   0.5263
ORTH           12       5        25       0.7059   0.3243   0.5714
OTHER          94       22       21       0.8103   0.8174   0.8117
PART           2        1        0        0.6667   1.0      0.7143
PREP           43       9        10       0.8269   0.8113   0.8238
PRON    

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {3}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M:DET          55       19       123      0.7432   0.309    0.5802
M:NOUN         9        3        2        0.75     0.8182   0.7627
M:OTHER        13       2        1        0.8667   0.9286   0.8784
M:PREP         7        3        3        0.7      0.7      0.7
M:PUNCT        10       0        1        1.0      0.9091   0.9804
M:VERB         9        1        0        0.9      1.0      0.9184
M:VERB:TENSE   2        1        1        0.6667   0.6667   0.6667
R:ADJ          11       3        1        0.7857   0.9167   0.8088
R:ADJ:FORM     0        3        1        0.0      0.0      0.0
R:ADV          2        0        0        1.0      1.0      1.0
R:CONJ         1        0        0        1.0      1.0      1.0
R:DET          10       5        6        0.6667   0.625    0.6579
R:MORPH        20       10       7        0.6667   0.7407   0.6803
R:NOUN  

## t5-lang8 + Llama

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt


=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
355	774	622	0.3144	0.3634	0.3231



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {1}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M              66       86       178      0.4342   0.2705   0.3873
R              260      580      398      0.3095   0.3951   0.3235
U              29       108      46       0.2117   0.3867   0.2327

=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
355	774	622	0.3144	0.3634	0.3231



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {2}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            15       11       6        0.5769   0.7143   0.6
ADJ:FORM       0        0        1        1.0      0.0      0.0
ADV            4        7        3        0.3636   0.5714   0.3922
CONJ           2        2        1        0.5      0.6667   0.5263
CONTR          0        1        0        0.0      1.0      0.0
DET            49       125      180      0.2816   0.214    0.2649
MORPH          8        29       13       0.2162   0.381    0.2367
NOUN           49       104      61       0.3203   0.4455   0.3393
NOUN:NUM       16       22       36       0.4211   0.3077   0.3922
ORTH           7        102      34       0.0642   0.1707   0.0734
OTHER          72       135      58       0.3478   0.5538   0.3758
PART           0        3        0        0.0      1.0      0.0
PREP           24       91       28       0.2087   0.4615   0.2344
PRON    

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {3}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M:ADJ          0        1        0        0.0      1.0      0.0
M:ADV          1        0        0        1.0      1.0      1.0
M:DET          28       39       143      0.4179   0.1637   0.3189
M:NOUN         9        4        5        0.6923   0.6429   0.6818
M:OTHER        13       8        7        0.619    0.65     0.625
M:PREP         4        22       6        0.1538   0.4      0.1754
M:PRON         0        2        1        0.0      0.0      0.0
M:PUNCT        10       4        5        0.7143   0.6667   0.7042
M:VERB         0        3        9        0.0      0.0      0.0
M:VERB:TENSE   1        3        2        0.25     0.3333   0.2632
R:ADJ          15       9        6        0.625    0.7143   0.641
R:ADJ:FORM     0        0        1        1.0      0.0      0.0
R:ADV          1        6        2        0.1429   0.3333   0.1613
R:CONTR      

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds


============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
396	733	581	0.3508	0.4053	0.3605



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {1}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M              70       83       174      0.4575   0.2869   0.4089
R              294      550      364      0.3483   0.4468   0.3644
U              32       100      43       0.2424   0.4267   0.2653

============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
396	733	581	0.3508	0.4053	0.3605



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {2}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            15       11       6        0.5769   0.7143   0.6
ADJ:FORM       0        0        1        1.0      0.0      0.0
ADV            4        6        3        0.4      0.5714   0.4255
CONJ           2        2        1        0.5      0.6667   0.5263
CONTR          0        1        0        0.0      1.0      0.0
DET            53       113      176      0.3193   0.2314   0.2968
MORPH          10       28       11       0.2632   0.4762   0.289
NOUN           55       98       55       0.3595   0.5      0.3809
NOUN:NUM       18       21       34       0.4615   0.3462   0.4327
ORTH           11       99       30       0.1      0.2683   0.1143
OTHER          82       123      48       0.4      0.6308   0.4316
PART           0        3        0        0.0      1.0      0.0
PREP           27       87       25       0.2368   0.5192   0.2657
PRON     

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {3}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M:ADJ          0        1        0        0.0      1.0      0.0
M:ADV          1        0        0        1.0      1.0      1.0
M:DET          29       38       142      0.4328   0.1696   0.3303
M:NOUN         9        4        5        0.6923   0.6429   0.6818
M:OTHER        15       6        5        0.7143   0.75     0.7212
M:PREP         4        22       6        0.1538   0.4      0.1754
M:PRON         0        2        1        0.0      0.0      0.0
M:PUNCT        10       4        5        0.7143   0.6667   0.7042
M:VERB         1        3        8        0.25     0.1111   0.2
M:VERB:TENSE   1        3        2        0.25     0.3333   0.2632
R:ADJ          15       9        6        0.625    0.7143   0.641
R:ADJ:FORM     0        0        1        1.0      0.0      0.0
R:ADV          1        5        2        0.1667   0.3333   0.1852
R:CONTR     

## t5-lang8 + REALEC

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt


=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
12177	11789	13362	0.5081	0.4768	0.5015



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds


============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
14295	9674	11244	0.5964	0.5597	0.5887



## t5-lang8 + GED + GEC + REALEC

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt


=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
613	185	343	0.7682	0.6412	0.7389



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {1}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M              172      52       64       0.7679   0.7288   0.7597
R              400      116      245      0.7752   0.6202   0.7383
U              41       17       34       0.7069   0.5467   0.6678

=========== Span-Based Correction ============
TP	FP	FN	Prec	Rec	F0.5
613	185	343	0.7682	0.6412	0.7389



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {2}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            12       3        1        0.8      0.9231   0.8219
ADJ:FORM       0        2        1        0.0      0.0      0.0
ADV            7        1        1        0.875    0.875    0.875
CONJ           2        0        0        1.0      1.0      1.0
DET            144      48       91       0.75     0.6128   0.7178
MORPH          24       5        3        0.8276   0.8889   0.8392
NOUN           66       29       25       0.6947   0.7253   0.7006
NOUN:INFL      1        0        0        1.0      1.0      1.0
NOUN:NUM       19       13       23       0.5938   0.4524   0.5588
ORTH           14       7        23       0.6667   0.3784   0.5785
OTHER          101      19       14       0.8417   0.8783   0.8487
PART           1        1        1        0.5      0.5      0.5
PREP           43       14       10       0.7544   0.8113   0.7651
PRON     

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -cat {3}


===================== Span-Based Correction ======================
Category       TP       FP       FN       P        R        F0.5
M:DET          124      35       54       0.7799   0.6966   0.7617
M:NOUN         8        4        3        0.6667   0.7273   0.678
M:OTHER        13       2        1        0.8667   0.9286   0.8784
M:PREP         6        6        4        0.5      0.6      0.5172
M:PUNCT        11       0        0        1.0      1.0      1.0
M:VERB         8        1        1        0.8889   0.8889   0.8889
M:VERB:TENSE   2        4        1        0.3333   0.6667   0.3704
R:ADJ          11       2        1        0.8462   0.9167   0.8594
R:ADJ:FORM     0        2        1        0.0      0.0      0.0
R:ADV          2        0        0        1.0      1.0      1.0
R:CONJ         1        0        0        1.0      1.0      1.0
R:DET          8        4        8        0.6667   0.5      0.625
R:MORPH        24       5        3        0.8276   0.8889   0.8392
R:NOUN    

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds


============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
621	177	335	0.7782	0.6496	0.7486



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {1}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M              173      51       63       0.7723   0.7331   0.7641
R              406      111      239      0.7853   0.6295   0.7482
U              42       15       33       0.7368   0.56     0.6931

============ Span-Based Detection ============
TP	FP	FN	Prec	Rec	F0.5
621	177	335	0.7782	0.6496	0.7486



In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {2}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
ADJ            12       3        1        0.8      0.9231   0.8219
ADJ:FORM       0        2        1        0.0      0.0      0.0
ADV            7        1        1        0.875    0.875    0.875
CONJ           2        0        0        1.0      1.0      1.0
DET            145      46       90       0.7592   0.617    0.7257
MORPH          24       5        3        0.8276   0.8889   0.8392
NOUN           66       29       25       0.6947   0.7253   0.7006
NOUN:INFL      1        0        0        1.0      1.0      1.0
NOUN:NUM       19       13       23       0.5938   0.4524   0.5588
ORTH           14       6        23       0.7      0.3784   0.5983
OTHER          102      17       13       0.8571   0.887    0.8629
PART           2        1        0        0.6667   1.0      0.7143
PREP           43       14       10       0.7544   0.8113   0.7651
PRON  

In [ ]:
!errant_compare -hyp /content/preds_m2.txt -ref /content/source_m2.txt -ds -cat {3}


====================== Span-Based Detection ======================
Category       TP       FP       FN       P        R        F0.5
M:DET          124      35       54       0.7799   0.6966   0.7617
M:NOUN         8        4        3        0.6667   0.7273   0.678
M:OTHER        14       1        0        0.9333   1.0      0.9459
M:PREP         6        6        4        0.5      0.6      0.5172
M:PUNCT        11       0        0        1.0      1.0      1.0
M:VERB         8        1        1        0.8889   0.8889   0.8889
M:VERB:TENSE   2        4        1        0.3333   0.6667   0.3704
R:ADJ          11       2        1        0.8462   0.9167   0.8594
R:ADJ:FORM     0        2        1        0.0      0.0      0.0
R:ADV          2        0        0        1.0      1.0      1.0
R:CONJ         1        0        0        1.0      1.0      1.0
R:DET          9        4        7        0.6923   0.5625   0.6618
R:MORPH        24       5        3        0.8276   0.8889   0.8392
R:NOUN   

## t5-lang8 + GED + GEC + Llama

=========== Span-Based Correction ============

TP	FP	FN	Prec	Rec	F0.5

13892	9715	11627	0.5885	0.5444	0.5791

==============================================

===================== Span-Based Correction ======================


Category       TP       FP       FN       P        R        F0.5

M              1215     664      969      0.6466   0.5563   0.6263

R              11522    8492     9892     0.5757   0.5381   0.5678

U              1155     559      766      0.6739   0.6012   0.658


=========== Span-Based Correction ============

TP	FP	FN	Prec	Rec	F0.5

13892	9715	11627	0.5885	0.5444	0.5791

==============================================

===================== Span-Based Correction ======================

Category       TP       FP       FN       P        R        F0.5

ADJ            381      208      301      0.6469   0.5587   0.6271

ADJ:FORM       5        3        6        0.625    0.4545   0.5814

ADV            96       86       149      0.5275   0.3918   0.4933

CONJ           59       36       66       0.6211   0.472    0.5842

DET            1325     461      520      0.7419   0.7182   0.737

MORPH          129      44       71       0.7457   0.645    0.7231

NOUN           2088     1195     1495     0.636    0.5828   0.6246

NOUN:INFL      4        0        0        1.0      1.0      1.0

NOUN:NUM       230      82       78       0.7372   0.7468   0.7391

NOUN:POSS      0        0        1        1.0      0.0      0.0

ORTH           120      33       37       0.7843   0.7643   0.7802

OTHER          6596     5982     6881     0.5244   0.4894   0.517

PART           40       29       36       0.5797   0.5263   0.5682

PREP           1028     600      772      0.6314   0.5711   0.6184

PRON           110      82       80       0.5729   0.5789   0.5741

PUNCT          22       16       32       0.5789   0.4074   0.534

SPELL          676      158      216      0.8106   0.7578   0.7994

VERB           566      470      619      0.5463   0.4776   0.5311

VERB:FORM      89       48       56       0.6496   0.6138   0.6421

VERB:INFL      13       3        3        0.8125   0.8125   0.8125

VERB:SVA       100      47       37       0.6803   0.7299   0.6897

VERB:TENSE     179      115      133      0.6088   0.5737   0.6015

WO             36       17       38       0.6792   0.4865   0.6294

=========== Span-Based Correction ============

TP	FP	FN	Prec	Rec	F0.5

13892	9715	11627	0.5885	0.5444	0.5791

==============================================


===================== Span-Based Correction ======================


Category       TP       FP       FN       P        R        F0.5

M:ADJ          22       16       38       0.5789   0.3667   0.5189

M:ADV          12       10       29       0.5455   0.2927   0.4651

M:CONJ         26       18       29       0.5909   0.4727   0.5628

M:DET          646      166      169      0.7956   0.7926   0.795

M:NOUN         77       78       116      0.4968   0.399    0.4736

M:OTHER        227      223      353      0.5044   0.3914   0.4769

M:PART         1        1        0        0.5      1.0      0.5556

M:PREP         134      68       107      0.6634   0.556    0.6387

M:PRON         9        16       10       0.36     0.4737   0.3782

M:PUNCT        6        7        11       0.4615   0.3529   0.4348

M:VERB         41       40       77       0.5062   0.3475   0.4638

M:VERB:FORM    1        2        0        0.3333   1.0      0.3846

M:VERB:TENSE   13       19       30       0.4062   0.3023   0.3801

R:ADJ          330      182      247      0.6445   0.5719   0.6286

R:ADJ:FORM     5        3        6        0.625    0.4545   0.5814

R:ADV          57       53       86       0.5182   0.3986   0.4889

R:CONJ         12       5        12       0.7059   0.5      0.6522

R:DET          419      195      275      0.6824   0.6037   0.6651

R:MORPH        129      44       71       0.7457   0.645    0.7231

R:NOUN         1904     1057     1283     0.643    0.5974   0.6334

R:NOUN:INFL    4        0        0        1.0      1.0      1.0

R:NOUN:NUM     230      82       78       0.7372   0.7468   0.7391

R:NOUN:POSS    0        0        1        1.0      0.0      0.0

R:ORTH         120      33       37       0.7843   0.7643   0.7802

R:OTHER        5971     5542     6185     0.5186   0.4912   0.5129

R:PART         32       24       33       0.5714   0.4923   0.5536

R:PREP         684      461      575      0.5974   0.5433   0.5857

R:PRON         77       46       50       0.626    0.6063   0.622

R:PUNCT        5        2        8        0.7143   0.3846   0.6098

R:SPELL        676      158      216      0.8106   0.7578   0.7994

R:VERB         478      407      502      0.5401   0.4878   0.5288

R:VERB:FORM    86       46       56       0.6515   0.6056   0.6418

R:VERB:INFL    13       3        3        0.8125   0.8125   0.8125

R:VERB:SVA     100      47       37       0.6803   0.7299   0.6897

R:VERB:TENSE   154      85       93       0.6444   0.6235   0.6401

R:WO           36       17       38       0.6792   0.4865   0.6294

U:ADJ          29       10       16       0.7436   0.6444   0.7214

U:ADV          27       23       34       0.54     0.4426   0.5172

U:CONJ         21       13       25       0.6176   0.4565   0.5769

U:DET          260      100      76       0.7222   0.7738   0.732

U:NOUN         107      60       96       0.6407   0.5271   0.6142

U:OTHER        398      217      343      0.6472   0.5371   0.6217

U:PART         7        4        3        0.6364   0.7      0.6481

U:PREP         210      71       90       0.7473   0.7      0.7374

U:PRON         24       20       20       0.5455   0.5455   0.5455

U:PUNCT        11       7        13       0.6111   0.4583   0.5729

U:VERB         47       23       40       0.6714   0.5402   0.6403

U:VERB:FORM    2        0        0        1.0      1.0      1.0

U:VERB:TENSE   12       11       10       0.5217   0.5455   0.5263



=========== Span-Based Correction ============

TP	FP	FN	Prec	Rec	F0.5

13892	9715	11627	0.5885	0.5444	0.5791

==============================================


============ Span-Based Detection ============

TP	FP	FN	Prec	Rec	F0.5

16773	6832	8746	0.7106	0.6573	0.6992

==============================================

====================== Span-Based Detection ======================

Category       TP       FP       FN       P        R        F0.5

M              1377     495      807      0.7356   0.6305   0.7118

R              14134    5864     7280     0.7068   0.66     0.6969

U              1262     473      659      0.7274   0.6569   0.7121


============ Span-Based Detection ============

TP	FP	FN	Prec	Rec	F0.5

16773	6832	8746	0.7106	0.6573	0.6992

==============================================


====================== Span-Based Detection ======================

Category       TP       FP       FN       P        R        F0.5

ADJ            450      139      232      0.764    0.6598   0.7406

ADJ:FORM       5        2        6        0.7143   0.4545   0.641

ADV            115      59       130      0.6609   0.4694   0.6111

CONJ           68       26       57       0.7234   0.544    0.6786

DET            1412     355      433      0.7991   0.7653   0.7921

MORPH          146      23       54       0.8639   0.73     0.8333

NOUN           2525     791      1058     0.7615   0.7047   0.7494

NOUN:INFL      4        0        0        1.0      1.0      1.0

NOUN:NUM       236      61       72       0.7946   0.7662   0.7888

NOUN:POSS      0        0        1        1.0      0.0      0.0

ORTH           126      22       31       0.8514   0.8025   0.8411

OTHER          8299     4263     5178     0.6606   0.6158   0.6512

PART           45       26       31       0.6338   0.5921   0.625

PREP           1176     445      624      0.7255   0.6533   0.7098

PRON           124      66       66       0.6526   0.6526   0.6526

PUNCT          28       10       26       0.7368   0.5185   0.6796

SPELL          806      74       86       0.9159   0.9036   0.9134

VERB           741      309      444      0.7057   0.6253   0.688

VERB:FORM      102      30       43       0.7727   0.7034   0.7578

VERB:INFL      16       1        0        0.9412   1.0      0.9524

VERB:SVA       111      29       26       0.7929   0.8102   0.7963

VERB:TENSE     202      85       110      0.7038   0.6474   0.6918

WO             36       16       38       0.6923   0.4865   0.6383


============ Span-Based Detection ============

TP	FP	FN	Prec	Rec	F0.5

16773	6832	8746	0.7106	0.6573	0.6992

==============================================


====================== Span-Based Detection ======================

Category       TP       FP       FN       P        R        F0.5

M:ADJ          24       13       36       0.6486   0.4      0.5769


M:ADV          14       7        27       0.6667   0.3415   0.56

M:CONJ         30       15       25       0.6667   0.5455   0.6383

M:DET          662      139      153      0.8265   0.8123   0.8236

M:NOUN         90       64       103      0.5844   0.4663   0.5562

M:OTHER        323      131      257      0.7115   0.5569   0.674

M:PART         1        1        0        0.5      1.0      0.5556

M:PREP         143      60       98       0.7044   0.5934   0.679

M:PRON         10       14       9        0.4167   0.5263   0.4348

M:PUNCT        6        6        11       0.5      0.3529   0.4615

M:VERB         56       32       62       0.6364   0.4746   0.5957

M:VERB:FORM    1        1        0        0.5      1.0      0.5556

M:VERB:TENSE   17       12       26       0.5862   0.3953   0.5346

R:ADJ          392      118      185      0.7686   0.6794   0.7489

R:ADJ:FORM     5        2        6        0.7143   0.4545   0.641

R:ADV          72       35       71       0.6729   0.5035   0.6305

R:CONJ         13       4        11       0.7647   0.5417   0.7065

R:DET          475      136      219      0.7774   0.6844   0.7569

R:MORPH        146      23       54       0.8639   0.73     0.8333

R:NOUN         2302     678      885      0.7725   0.7223   0.7619

R:NOUN:INFL    4        0        0        1.0      1.0      1.0

R:NOUN:NUM     236      61       72       0.7946   0.7662   0.7888

R:NOUN:POSS    0        0        1        1.0      0.0      0.0

R:ORTH         126      22       31       0.8514   0.8025   0.8411

R:OTHER        7554     3928     4602     0.6579   0.6214   0.6503

R:PART         37       21       28       0.6379   0.5692   0.6229

R:PREP         807      329      452      0.7104   0.641    0.6953

R:PRON         86       34       41       0.7167   0.6772   0.7084

R:PUNCT        10       1        3        0.9091   0.7692   0.8772

R:SPELL        806      74       86       0.9159   0.9036   0.9134

R:VERB         630      257      350      0.7103   0.6429   0.6957

R:VERB:FORM    99       29       43       0.7734   0.6972   0.7569

R:VERB:INFL    16       1        0        0.9412   1.0      0.9524

R:VERB:SVA     111      29       26       0.7929   0.8102   0.7963

R:VERB:TENSE   171      66       76       0.7215   0.6923   0.7155

R:WO           36       16       38       0.6923   0.4865   0.6383

U:ADJ          34       8        11       0.8095   0.7556   0.7981

U:ADV          29       17       32       0.6304   0.4754   0.5918

U:CONJ         25       7        21       0.7812   0.5435   0.7184

U:DET          275      80       61       0.7746   0.8185   0.783

U:NOUN         133      49       70       0.7308   0.6552   0.7143

U:OTHER        422      204      319      0.6741   0.5695   0.6502

U:PART         7        4        3        0.6364   0.7      0.6481

U:PREP         226      56       74       0.8014   0.7533   0.7913

U:PRON         28       18       16       0.6087   0.6364   0.614

U:PUNCT        12       3        12       0.8      0.5      0.7143

U:VERB         55       20       32       0.7333   0.6322   0.7106

U:VERB:FORM    2        0        0        1.0      1.0      1.0

U:VERB:TENSE   14       7        8        0.6667   0.6364   0.6604

============ Span-Based Detection ============

TP	FP	FN	Prec	Rec	F0.5

16773	6832	8746	0.7106	0.6573	0.6992

==============================================